### ISSUE TITLE
Rule proposal: `no-single-promise-in-promise-methods`

### ISSUE THREAD

**Issue by Clement398** [sindresorhus/eslint-plugin-unicorn#2256]
### Description

Disallow using `Promise` method with a single element array as parameter.

### Fail

```js
Promise.all([promise]);

Promise.any([promise]);

Promise.race([promise]);
```

### Pass

```js
Promise.all([promise, anotherPromise]);
Promise.all(notArrayLiteral);
Promise.all([...promises]);

Promise.any([promise, anotherPromise]);

Promise.race([promise, anotherPromise]);

Promise.allSettled([promise]);
```

**Comment by sindresorhus** [comment]
This could be more general and also apply to `Promise.allSettled` and `Promise.race`.

**Comment by Clement398** [comment]
It's edited, thanks for the suggestion.
